File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    let answer = '';
    const map = new Map();
    
    completion.forEach((value) => {
        map.set(value, map.get(value) + 1 || 1);
    })
    participant.some((value) => {
        if(map.get(value) >= 1){
            map.set(value, map.get(value)-1);
            return false;
        }
        answer = value;
        return false;
    })
    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    p_m = {}
    for p in participant:
        p_m[p] = p_m.get(p, 0) + 1
    for c in completion:
        p_m[c] -= 1
    for k, v in p_m.items():
        if v > 0:
            return k

